src/server.js

```diff
@@ -55,38 +34,43 @@ function link(res, path, rel, forPath, headers) {
   if (!headers['Link']) {
     headers['Link'] = [];
   }
-  headers['Link'].push(`<${path}>; rel=${rel}; as=script`);
+  headers['Link'].push(`<${path}>; rel=${rel}; as=script nopush`);
 }
 
 
 // Request handler.
 function onRequest(req, res) {
+  const dirPath = '../public'
   const [reqPath, query] = req.url.split('?');
   const filePath = reqPath == '/' ? '/index.html' : reqPath;
   console.log('onRequest: ', req.url, reqPath, query || '',
       req.headers['cache-control']);
-  const file = publicFiles.get(filePath);
+
+  const file = fs.readFileSync(dirPath + filePath);
+  const stat = fs.statSync(dirPath + filePath)
 
   // File not found
-  if (!file) {
+  if (!file || !stat) {
     console.log('404 ', reqPath);
     res.statusCode = 404;
     res.end();
     return;
   }
 
   // Push or link if needed.
-  const headers = {};
-  if (reqPath.endsWith('.js')) {
-    headers['Cache-Control'] = 'public, max-age=600';
-  }
+  const headers = {
+    'content-length': stat.size,
+    'last-modified': stat.mtime.toUTCString(),
+    'Cache-Control': 'public, max-age=600',
+    'Content-Type': 'application/javascript',
```

Comment:
What if it's not a JS? The html files are here as well.

Reply:
The middle ware check for 
``
app.get('*.js/', (req, res) => {
  onRequest(req, res);
});
```
It only applies to JS file

Reply:
I see. OK. It works then. I wanted to add some additional massaging for html files, but we can go w/o for now.

Reply:
Got it. we can easily support that, `express.static()` allows us to setHeader to files easily.